### PR TITLE
[TTAHUB-2811] Resolve XSS 

### PR DESCRIPTION
**Pull request by thewatermethod**
## Description of change
This is a redo of @nvms's #2089. That had a messy history with most of it being reverted due to a bug introduced prior (the bug this very PR was intended to fix).

Use DOMPurify ([github page](https://github.com/cure53/DOMPurify)) ([npm page](https://www.npmjs.com/package/dompurify)) to sanitize rich fields.

## How to test
Put a bunch of formatted HTML into one of the rich text fields. Save to the database. Ensure that the stored data is completely sanitized. Reload the rich text field, and validate that what is returned to the client doesn't include &lt;, &gt;, etc.

## Issue(s)

* https://jira.acf.gov/browse/TTAHUB-2811

### Every PR

- [ ] Meets issue criteria
- [ ] JIRA ticket status updated
- [ ] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Before merge to main

- [ ] OHS demo complete
- [ ] Ready to create production PR

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

File: src/models/helpers/escapeFields.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+/**
+ * Escape fields in sequelize instance
+ * Intended to be user in sequelize hooks "beforeCreate" and "beforeUpdate"
+ *
+ * @param {Object} instance - sequelize model instance
+ * @param {String[]} fields - array of fields to escape
+ * @returns void
+ */
+function escapeFields(instance, fields) {
```
Comment (AdamAdHocTeam) [inline src/models/helpers/escapeFields.js:44]: Correct me if I'm wrong we are no longer escaping fields using lodash. We should probably rename this function to purify fields (also anywhere else we use the escape language).

File: src/models/hooks/activityReport.js [hunk 2]
```diff
@@ -23,6 +23,9 @@ const {
   removeCollaboratorsForType,
 } = require('../helpers/genericCollaborator');
 const { destroyLinkedSimilarityGroups } = require('./activityReportGoal');
+const { purifyFields } = require('../helpers/purifyFields');
+
+const AR_FIELDS_TO_ESCAPE = ['additionalNotes', 'context'];
```
Comment (AdamAdHocTeam) [inline src/models/hooks/activityReport.js:28]: Should we also apply this to Objective TTA Provided as its also rich text (respectively)?

Comment (thewatermethod) [reply src/models/hooks/activityReport.js:28]: I think we should start with the fields flagged by the pen test

Comment (AdamAdHocTeam) [reply src/models/hooks/activityReport.js:28]: Ok lets create a ticket for it so its not lost.

**Review by AdamAdHocTeam — APPROVED** [review]
Looks good. Was able to create some formatting on rich text and add a url.

Left some minor comments (also wait for test to pass).

**Pull request merged**